dpks/quant_matrix.py

```diff
@@ -459,121 +461,205 @@ def compare_groups(
         scale: bool = True,
-        min_samples_per_group: int = 2,
-        feature_importance_method: str = "rfecv",
-        calculate_feature_importance: bool = False,
-        run_param_search: bool = False,
+        rank_method: str = "rfecv",
         **kwargs: Union[dict, int, str, bool],
-    ) -> QuantMatrix:
-        identifiers = self.proteins
+    ):
 
-        quant_copy = self.quantitative_data.copy()
-        quant_copy.X[quant_copy.X == 0.0] = np.nan
-        drop_indexes = []
-        groups = self.quantitative_data.var["group"].unique()
+        X = format_data(self)
+        y = encode_labels(self.quantitative_data.var["group"].values)
+
+        if scale:
+            if scaler:
+                X = scaler.fit_transform(X)
+            else:
+                scaler = StandardScaler()
+                X = scaler.fit_transform(X)
 
-        for group in groups:
-            for identifier in identifiers:
-                quant_data = quant_copy[
-                    self.row_annotations["Protein"] == identifier, :
-                ].copy()
+        verbose = bool(kwargs.get("verbose", False))
 
-                index = int(quant_data.obs.index.to_numpy()[0])
+        if rank_method == "rfecv":
 
-                group_data = quant_data[:, self.get_samples(group=group)].X.copy()
-                group_nonan = len(group_data[~np.isnan(group_data)])
+            rfe_step = int(kwargs.get("rfe_step", 1))
+            rfe_min_features_to_select = int(
+                kwargs.get("rfe_min_features_to_select", 1)
+            )
 
-                if group_nonan < min_samples_per_group:
-                    if index not in drop_indexes:
-                        drop_indexes.append(index)
+            k_folds = int(kwargs.get("k_folds", 2))
 
-        le = LabelEncoder()
-        Y = le.fit_transform(self.quantitative_data.var["group"].values)
-        X = self.quantitative_data.X.copy().transpose()
-        X = np.delete(X, drop_indexes, 1)
-        X = np.nan_to_num(X, copy=True, nan=0.0)
+            threads = int(kwargs.get("threads", 1))
+
+            selector = FeatureRankerRFE(
+                min_features_to_select=rfe_min_features_to_select,
+                step=rfe_step,
+                importance_getter=shap_algorithm,
+                scoring="accuracy",
+                k_folds=k_folds,
+                threads=threads,
+                verbose=verbose,
+            )
+
+            selector.rank_features(X, y, classifier)
+
+            feature_rank_values = selector.ranking_.tolist()
+
+            self.quantitative_data.obs["FeatureRank"] = feature_rank_values
+
+            self.selector = selector
+
+        return self
+
+    def predict(
+        self,
+        classifier,
+        scaler,
+        shap_algorithm: str = "auto",
+        scale: bool = True,
+    ) -> QuantMatrix:
+
+        X = format_data(self)
+        y = encode_labels(self.quantitative_data.var["group"].values)
 
         if scale:
```

Comment:
Here we should probably have same as above in the rank method. Scaler should be optional, default None, and then the if `scale == True` but `not scaler` we should add StandardScaler as default.